Sources/SwiftParser/Patterns.swift

```diff
@@ -120,20 +120,31 @@ extension Parser {
   /// =======
   ///
   ///     typed-pattern → pattern ':' attributes? inout? type
-  mutating func parseTypedPattern() -> (RawPatternSyntax, RawTypeAnnotationSyntax?) {
+  mutating func parseTypedPattern(allowRecover: Bool = true) -> (RawPatternSyntax, RawTypeAnnotationSyntax?) {
```

Comment:
Nitpick: We call this parameter `allowRecovery` in the rest of the codebase. But I think in this case, I would be more specific and call it `allowRecoveryFromMissingColon`.